### ISSUE TITLE
Prevent more common formats from transcoding when using -x

### ISSUE THREAD

**Issue by Arynthesizer** [pukkandan/yt-dlp#58]
## Checklist

- [x] I'm reporting a broken site support issue
- [x] I've verified that I'm running yt-dlp version **2021.02.04**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
C:\ytdl>youtube-dlc -v -x --audio-format "best" https://soundcloud.com/belikboom/harry-potter
[debug] Command-line config: ['-v', '-x', '--audio-format', 'best', 'https://soundcloud.com/belikboom/harry-potter']
[debug] Loading archive file None
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.02.04
[debug] Python version 3.8.7 (CPython) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-11-19-full_build-www.gyan.dev, ffprobe 4.3.1-2020-11-19-full_build-www.gyan.dev
[debug] Proxy map: {}
[soundcloud] belikboom/harry-potter: Downloading info JSON
[soundcloud] 172989719: Downloading JSON metadata
[soundcloud] 172989719: Downloading webpage
[soundcloud] 172989719: Downloading JSON metadata
[soundcloud] 172989719: Downloading JSON metadata
[soundcloud] 172989719: Downloading JSON metadata
[debug] Sort order given by user:
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9(9), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Downloading format(s) download
[debug] Invoking downloader on 'https://cf-media.sndcdn.com/WYHtGMNj90Yp?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vV1lIdEdNTmo5MFlwIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjEyNzQxMjk0fX19XX0_&Signature=SuNXZ5Op0HXNZEvZHZjA3S17XyK7eB9jhLv8YMrQ3mGZ-kyhlryWDvb5IpTKy07mBgTq-SV9W1rty6VwTyliVOYF8tYH74LaCs8e1Ya5ZswFrSiFmFwelWo5RqcLcxOqzKt7Z4zuyaySd7PntIa9m-BwiKR5L7E29PQOM9GwQzbsdRidC~Zlgapf-FZGY5AyyGN2tO40wz300bAe2VfX8btkq9AkLYdRyOioPI6eW40Vs085pr2Luzwbd6Pj9~MOEB0WKw7IePvl8af7R6zH0O3i7V0iHcQS01kLCft-V0~UPnvtr4ZV71IkKJuy7TWrtQxcCkP3~TLcquJuqa9pEw__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ'
[download] Destination: Avada kadabra (FREE DOWNLOAD) [172989719].wav
[download] 100% of 67.68MiB in 00:00
[debug] ffmpeg command line: ffprobe -show_streams "file:Avada kadabra (FREE DOWNLOAD) [172989719].wav"
[ExtractAudio] Destination: Avada kadabra (FREE DOWNLOAD) [172989719].mp3
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:Avada kadabra (FREE DOWNLOAD) [172989719].wav" -vn -acodec libmp3lame "-q:a" 5 "file:Avada kadabra (FREE DOWNLOAD) [172989719].mp3"
Deleting original file Avada kadabra (FREE DOWNLOAD) [172989719].wav (pass -k to keep)

```

## Description

Previously posted about here: https://github.com/blackjack4494/youtube-dlc/issues/214

Basically, the original wav downloads properly, but is then converted to a lower-quality 137 kbps mp3 despite the "best" argument. The command works as expected with songs native to other formats, for example it converts aac to m4a, which is why I still use it :)

Thanks!

**Comment by pukkandan** [comment]
This is the core issue: https://github.com/ytdl-org/youtube-dl/issues/28006#issuecomment-770315470

I will see what I can do to fix it

**Comment by pukkandan** [comment]
Can someone give me a list of known audio extensions?

**Comment by Arynthesizer** [comment]
This one from http://dotwhat.net/type/audio-music-files seems pretty comprehensive but I'm sure the vast majority are never used

.3GA - 3GPP Audio FileAverage
.AA - Audible Audio fileCommon
.AA3 - ATRAC3 Audio FileCommon
.AAC - Advanced Audio Coding FileVery Common
.AAX - Audible Enhanced AudiobookAverage
.ABC - Musical Notation Language Filenot specified
.AC3 - Dolby Digital Audio Codec 3 file formatAverage
.ACD - Acid Project FileAverage
.ACD-ZIP - Sony Acid Project File (Inc Embedded Data)Average
.ACM - InterPlay ACM Audio file formatAverage
.ACT - ADPCM Audio FileAverage
.ADG - Ableton Device GroupAverage
.ADTS - Audio Data Transport Stream file formatAverage
.AFC - Mass Effect 2 Audio FileAverage
.AHX - WinAHX Tracker Modulenot specified
.AIF - Audio Interchange FileVery Common
.AIFC - Audio Interchange File (Compressed)not specified
.AIFF - Audio Interchange FileVery Common
.AL - ALAW Format SoundAverage
.AMR - Adaptive Multi-Rate file formatCommon
.AMZ - Amazon MP3 Downloader Download FileCommon
.AOB - DVD-Audio Audio Object Filenot specified
.APC - Cryo APC Audio file formatUncommon
.APE - Monkeys Audio FileCommon
.APF - Sony Ericsson Acoustics Settings Filenot specified
.ATRAC - Sony ATRAC Audio Filenot specified
.AU - Sun Audio File FormatAverage
.AVR - Audio Visual Research filenot specified
.AWB - AMR-WB Audio Filenot specified
.AWB - Compressed Speech Audio Filenot specified
.BAP - Blaze Audio Wave Information Filenot specified
.BMW - BuzzPlay music with wavesAverage
.CAF - Core Audio FileCommon
.CDA - CD Audio Track ShortcutCommon
.CFA - Conformed Audio FileCommon
.CIDB - iTunes CD Information Filenot specified
.COPY - Sony Ericsson Protected Content Filenot specified
.CPR - Cubase Project FileAverage
.CWP - Cakewalk SONAR Projectnot specified
.DAC - DAC Sound FileAverage
.DCF - DRM Content Format FileAverage
.DCM - DCM Audio ModuleAverage
.DCT - Dictation Audio Filenot specified
.DFC - Defractor Instrumentnot specified
.DIG - Sound Designer Audio FileUncommon
.DSM - Digital Sound Modulenot specified
.DSS - Digital Speech Standard FileAverage
.DTS - Audio File FormatAverage
.DTSHD - DTS-HD Master AudioAverage
.DVF - Sony Compressed Audio Format Filenot specified
.EFA - Ensoniq ASR filenot specified
.EFE - Ensoniq EPS filenot specified
.EFK - Ensoniq KT filenot specified
.EFV - Ensoniq VFX-SD filenot specified
.EMD - ABT Extended MoDulenot specified
.EMX - eMusic Download Manager Download file formatCommon
.ENC - Encore Musical NotationAverage
.F64 - Raw 64-Bit Audio Filenot specified
.FL - Floating Sound FormatAverage
.FLAC - Free Lossless Audio Codec FileCommon
.FLP - Fruity Loops Project FileAverage
.FST - Studio State FileAverage
.G726 - G.726 Audio Filenot specified
.GNT - Letter-Art Filenot specified
.GPX - Guitar Pro 6 Document file formatCommon
.GSM - Audio stream Raw GSMnot specified
.GSM - eFax Messenger Plus Voice Messagenot specified
.HMA - Sony Connect (SonicStage) Audio FileAverage
.HTW - Hauptwerk Sound-fontAverage
.IFF - Interchange File FormatVery Common
.IKLAX - iKlax Multi-Track Audionot specified
.IMW - IncrediMail MIDI/WAVUncommon
.IMY - Ringtone FileAverage
.ITS - Impulse Tracker Samplenot specified
.IVC - Intervoice Audio FileAverage
.K26 - Kurzweil K2600 Filenot specified
.KAR - Karaoke MIDI FileAverage
.KFN - Audio Filenot specified
.KOE - AquesTone Vocal Synthesiser Syllable FileAverage
.KOZ - Audiokoz Music Filenot specified
.KOZ - Bell Music Filenot specified
.KPL - Kazaa Playlist FileAverage
.KTP - Kinetic Project Templatenot specified
.LQT - Liquid Audio Music Filenot specified
.LVP - Avaya Voice Player - Compressed Voice Audio Filenot specified
.M3U - Media Playlist FileVery Common
.M3U8 - MP3 Playlist File (UTF-8)Common
.M4A - MPEG-4 Audio LayerVery Common
.M4B - MPEG-4 Audio Book Filenot specified
.M4P - iTunes Audio Formatnot specified
.M4R - MPEG-4 Ringtone file formatCommon
.MA1 - Monarch Audio Filenot specified
.MID - Musical Instrument Digital Interface (MIDI) Sound FileVery Common
.MIDI - Musical Instrument Digital Interface (MIDI) Sound Filenot specified
.MINIUSF - Nintendo 64 Song FileUncommon
.MIO - MIO Compressed Audionot specified
.MKA - Matroska Audio file formatAverage
.MMF - Mobile Music FileAverage
.MON - Wavelab Audio Montage FileAverage
.MP2 - MPEG Audio StreamAverage
.MP3 - MP3 Audio FileVery Common
.MPA - MPEG Audio StreamVery Common
.MPC - Musepack Audio Filenot specified
.MPU - MPEG Layer 3 Audio Filenot specified
.MP_ - Mobile Phone Sound Filenot specified
.MSV - Sony Memory Stick Compressed Voice FileAverage
.MT2 - MadTracker 2 Modulenot specified
.MTE - MadTracker 2 Envelopenot specified
.MTP - MadTracker 2 Patternnot specified
.MUP - Music Publisher FileAverage
.MXP4 - Musinaut AudioAverage
.MZP - Mozart Percussion FileUncommon
.NCOR - Adobe Encore Project FileCommon
.NKI - KONTAKT Instrument FileCommon
.NRT - Nokia RingToneUncommon
.NSA - Nullsoft Streaming Audio Filenot specified
.NTN - NOTION Song Filenot specified
.NWC - Noteworthy Composer Song FileUncommon
.ODM - OverDrive Media Filenot specified
.OGA - Ogg Vorbis AudioVery Common
.OGG - Ogg Vorbis Codec Compressed WAV FileCommon
.OMA - Sony OpenMG Audio FileAverage
.OMG - OpenMG Audio Filenot specified
.OMX - OtsAV Media Library Information Filenot specified
.OTS - OtsAV Album Filenot specified
.OVE - Cakewalk Overture ScoreAverage
.PCAST - iTunes Podcast file formatCommon
.PEK - Adobe Peak Waveform FileCommon
.PLA - Playlist FileAverage
.PLS - PlayList FileCommon
.PNA - PhatNoise Audio Filenot specified
.PROG - Cakewalk Sonar Beatscape Preset Program FileUncommon
.PVC - Panasonic VM1 Voice Filenot specified
.QCP - Voice Filenot specified
.R1M - RealOne Streaming Media Filenot specified
.RA - RealMedia Streaming Audio FileVery Common
.RAM - RealMedia Metafilenot specified
.RAW - Raw Audio File Formatnot specified
.RAX - Real Music Store Audio Filenot specified
.REX - Audio Sample Filenot specified
.RFL - Reason ReFill Sound BankAverage
.RIF - Audio File in WAV formatUncommon
.RMJ - Real Media Jukebox Audio Filenot specified
.RNS - Reason Song FileCommon
.RSD - Roland Sound Filenot specified
.RSO - NXT Brick Audio Filenot specified
.RTI - Real Tracker Instrumentnot specified
.RX2 - REX2 Loop FileAverage
.SA1 - Audio Encryption FileAverage
.SBR - Spectral Band Replication Audio Sound FileAverage
.SD2 - Sound Designer 2 Flattened File / Data ForkAverage
.SFA - Sound Forge Audio FileCommon
.SGT - Microsoft DirectMusic Producer Segments FileUncommon
.SID - PlaySID C64 Sound Effects FileRare
.SMF - Standard Midi FileCommon
.SND - Apple Sound Resource FileCommon
.SNG - Midi song filenot specified
.SNS - Burnout Paradise Sound Filenot specified
.SPRG - Synclavier Program Filenot specified
.SSEQ - Synclavier Sequence Filenot specified
.SSND - Synclavier Sound Filenot specified
.SWA - ShockWave Audionot specified
.SYH - Synchomatic Instrumentnot specified
.SZ - Winamp Classic Skin DownloadAverage
.TAP - Wave Sound or Action WaveRare
.TRM - For The Record Voice Recording Filenot specified
.UL - Audio fileAverage
.USF - Nintendo 64 Music Filenot specified
.USFLIB - Nintendo 64 Song Librarynot specified
.USM - Unsigned Sample Module data FarandoleRare
.VAG - Sony PlayStation ADPCM Sound FileUncommon
.VMO - Siemens Mobile Phone Voice FileAverage
.VOI - Voyetra Voice Filenot specified
.VOX - Dialogic Voice FileAverage
.VPM - Garmin GPS Navigation Voice Filenot specified
.VRF - Ventrilo Audio Recordingnot specified
.VY1 - VY3 ETC - Audio Footage FileUncommon
.VYF - Samsung Digital Voice Recorder Filenot specified
.W01 - Yamaha SY Series Wave Filenot specified
.W64 - Wave64 FileAverage
.WAV - Waveform AudioVery Common
.WMA - Windows Media Audio FileVery Common
.WPROJ - Wwise ProjectAverage
.WRK - CakeWalk Project fileAverage
.WUS - WUTG Tagged Audio Filenot specified
.WUT - WUTG Tag Filenot specified
.WWU - Wwise Work Unitnot specified
.XFS - eRacer Sound Filenot specified
.ZGR - BeatCreator Loop FileAverage
.ZVR - SAFA Audio Recording File

**Comment by pukkandan** [comment]
Let me rephrase
What I need is a list of audio formats that is commonly used and should be kept as-is.

For example, while `aac` is a valid audio extension, the user probably expects `m4a` when using the aac codec

**Comment by Arynthesizer** [comment]
Ah, my mistake

From my recollection, 
`wav`
`flac`
`m4a`
`aiff`
`mp3`
`ogg`
`mka`
`opus`
`wma`

Agreed about `aac` -> `m4a`. Some would argue uncompressed lossless like `wav` or `aiff` should always be compressed to `flac` to reduce file size, but others probably want the files uncompressed for compatibility reasons.

**Comment by pukkandan** [comment]
> Some would argue uncompressed lossless like wav or aiff should always be compressed to flac to reduce file size, but others probably want the files uncompressed for compatibility reasons.

hm.. It's probably best to create a more advanced syntax for `--audio-format` similar to the one for `--remux-video` so that this concern can be addressed.

But for the time-being, I'm moving ahead with the list you gave

**Comment by Arynthesizer** [comment]
Hey, sorry to bother again. I've updated to 2021.02.15, and there is a message saying "Skipping audio extraction since the file is already in a common audio format", yet it transcodes anyway.

```
Z:\ytdl>youtube-dlc -v -x https://soundcloud.com/audyssey1/way-back
[debug] Command-line config: ['-v', '-x', 'https://soundcloud.com/audyssey1/way-back']
[debug] Loading archive file None
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.02.15 (exe)
[debug] Python version 3.8.7 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.3.1-2020-11-19-full_build-www.gyan.dev, ffprobe 4.3.1-2020-11-19-full_build-www.gyan.dev
[debug] Proxy map: {}
[soundcloud] audyssey1/way-back: Downloading info JSON
[soundcloud] 458940162: Downloading JSON metadata
[soundcloud] 458940162: Downloading webpage
[soundcloud] 458940162: Downloading JSON metadata
[soundcloud] 458940162: Downloading JSON metadata
[soundcloud] 458940162: Downloading JSON metadata
[debug] Sort order given by user:
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9(9), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Downloading format(s) download
[debug] Invoking downloader on 'https://cf-media.sndcdn.com/vTyzXuLoL9Kc?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLW1lZGlhLnNuZGNkbi5jb20vdlR5elh1TG9MOUtjIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjEzNDMzMzUxfX19XX0_&Signature=eQlxSHJSAyaobFOJMUnLHZWJpjYHDmvdiZgP09N5kFh2Z~thaqkX8ls59IG1-Fsvh2tPJxgIUnuXkYrxlhrYRvKMomJPpQxGUFDkJl82dksLIfbB169WQW-HK0HDh5AWFs5k~71p1SNTU~7jQubjr3PBh7icRe0~PxxNGN5mdUB45fAdc3UQ2VZXLNVtNJ6UkUsSiQM3ZnWugI-z0tz1AWMZjE6G17yLlM8N20HW1aKTUoaT3z8N0AeYTFqXaF2W9-H6~WfN6MDaI4on-o7AC8vE02NrYbbQLBjKU2ycSjwcVTTnzSGzu2IyirxaeFS4b40k8Iw50m1AvpP3aW1wFg__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ'
[download] Destination: Vicetone - Way Back (feat. Cozi Zuehlsdorff) - (Audyssey Remix) [458940162].wav
[download] 100% of 45.47MiB in 00:00
[ExtractAudio] Skipping audio extraction since the file is already in a common audio format
[debug] ffmpeg command line: ffprobe -show_streams "file:Vicetone - Way Back (feat. Cozi Zuehlsdorff) - (Audyssey Remix) [458940162].wav"
[ExtractAudio] Destination: Vicetone - Way Back (feat. Cozi Zuehlsdorff) - (Audyssey Remix) [458940162].mp3
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:Vicetone - Way Back (feat. Cozi Zuehlsdorff) - (Audyssey Remix) [458940162].wav" -vn -acodec libmp3lame "-q:a" 5 "file:Vicetone - Way Back (feat. Cozi Zuehlsdorff) - (Audyssey Remix) [458940162].mp3"
Deleting original file Vicetone - Way Back (feat. Cozi Zuehlsdorff) - (Audyssey Remix) [458940162].wav (pass -k to keep)
```

**Comment by pukkandan** [comment]
oops. That was a dumb mistake on my part 🤦

Fixed now
